File: code_examples/3-Structured_Problem_Solving/16-Break/Java/Main.java
/*
 * This program uses a break statement
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    // this function uses a break statement

    // create Scanner object for user input
    Scanner scanner = new Scanner(System.in);

    // input
    System.out.print("Enter how many times to repeat: ");
    String positiveIntegerStr = scanner.nextLine();
    System.out.println();

    // process & output
    int positiveInteger = Integer.parseInt(positiveIntegerStr);

    for (int loopCounter = 0; loopCounter < positiveInteger; loopCounter++) {
      System.out.println("%d time(s) through the loop.".formatted(loopCounter));
      if (loopCounter >= 5) {
        break;
      }
    }

    // close the Scanner object
    scanner.close();
    System.out.println("\nDone.");
  }
}

File: code_examples/3-Structured_Problem_Solving/16-Break/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module uses a break statement
"""


def main() -> None:
    """The main() function uses a break statement, returns None."""

    # input
    positive_integer = int(input("Enter how many times to repeat: "))
    print("")

    # process & output
    for loop_counter in range(positive_integer):
        print(f"{loop_counter} time through loop.")
        if loop_counter >= 5:
            break

    print("\nDone.")


if __name__ == "__main__":
    main()
